File: Ruby/Deviations/ListDeviations.rb
require 'net/http'
require 'uri'
require 'json'

boards_on_fire_api_url = "YOUR URL"  # example: https://my-company.boardsonfireapp.com/api/v5/
api_key = "YOUR API KEY"

uri = URI.parse("#{boards_on_fire_api_url}/deviations/list")
request = Net::HTTP::Post.new(uri)
request["Content-Type"] = "application/json"
request["x-api-key"] = api_key

request_body = {
  page: 1,
  page_size: 10
}

request.body = request_body.to_json

response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
  http.request(request)
end

if response.is_a?(Net::HTTPSuccess)
  data = JSON.parse(response.body)
  puts "Deviations:"
  puts data
else
  puts "Request error: #{response.message}"
end

File: Python/Deviations/ListDeviations.py
import requests
import json

boards_on_fire_api_url = "YOUR URL"  # example: https://my-company.boardsonfireapp.com/api/v5/
api_key = "YOUR API KEY"

headers = {
    "Content-Type": "application/json",
    "x-api-key": api_key
}

request_body = {
    "page": 1,
    "page_size": 10
}

try:
    response = requests.post(f"{boards_on_fire_api_url}/deviations/list", headers=headers, data=json.dumps(request_body))
    response.raise_for_status()
    data = response.json()
    print("Deviations:")
    print(data)
except requests.exceptions.RequestException as e:
    print(f"Request error: {e}")
